Automattic/woocommerce-payments
# Fix currency name not translated in the Overview card title

Fixes #3593

#### Changes proposed in this Pull Request

The currency name in the Overview card title is not translated; it's always in english.

The problem is that the translation strings extractor does not support variables:
https://github.com/Automattic/woocommerce-payments/blob/7e5c7186df90081ccbd2f8077387b30cd7ddf063/client/utils/currency/index.js#L34-L35

This PR moves the `__()` function call from the `formatCurrencyName()` function to the `currencyNames` initialization.

#### Testing instructions
0. Checkout the `develop` branch.
1. Go to **Settings** and change the **Site Language** to any supported languages (for example Spanish).
2. Go to **Dashboard > Updates** and click **Update translations** (this will be shown in the selected language).
3. We need to manually add the new strings to the language pack... Edit the file with the hash `090b05cdc4c33b5b4d778c13de44d8ea` in the filename (that's the json that corresponds to `dist/index.js`) and add the translation for **United States (US) dollar**:
   - For example for Spanish edit the file `wordpress/wp-content/languages/plugins/woocommerce-payments-es_ES-090b05cdc4c33b5b4d778c13de44d8ea.json`
   - Append the new translation at the end:
`... ,"Payments":["Pagos"]}}}` should become: `... ,"Payments":["Pagos"],"United States (US) dollar":["Dólar Estados Unidos"]}}}`
4. Go to **Payments > Overview** and verify that the currency name in the card title is not translated:
<img width="742" alt="147982731-93ab4c03-9c68-4836-8e6c-9f756c1befb4-2" src="https://user-images.githubusercontent.com/407542/150003503-f5c08a79-34d8-4147-af11-73b92c06414b.png">
5. Checkout this branch `fix/3593-currency-name-not-translated`.
6. Go to **Payments > Overview** again and verify that this time the currency name in the card title is correctly translated:
<img width="716" alt="Screen Shot 2022-01-18 at 16 22 30" src="https://user-images.githubusercontent.com/407542/150004485-243f56d2-3c88-492a-a906-3e97be0988f6.png">

__NOTE__: We need to manually edit the language pack (JSON file) to test this because the new strings are extracted from develop, after merging this PR the currency names will be added to the available strings to be translated (and sent for translation) when the weekly 18n release is generated. 

---
